### PR TITLE
SDK - Add Permit and Permit2 content

### PR DESCRIPTION
**Pull request by brunoguerios**

### CODE REVIEW

**Comment by brunoguerios** [comment]
Prettier applied a bunch of changes and made it a bit hard to follow actual changes I made 😕 
Here's a list to make the review easier:
- add buildCallWithPermit2 to addLiquidity section
- add buildCallWithPermit to removeLiquidity section
- add buildCallWithPermit2 to swap section
- add Permit2 Helper section
- add Permit Helper section
- some other minor fixes

Ps: I'm wondering why prettier auto-formatting was not applied before - how is this setup on your end?

**Comment by MattPereira** [comment]
> Ps: I'm wondering why prettier auto-formatting was not applied before - how is this setup on your end?

my settings match yours so i can save this PR file w/o changes, but i have run into same problem in the past when editing the create a custom AMM page where i ended up saving w/o formatting to avoid rocking the boat :sweat_smile:

**Review by johngrantuk — APPROVED** [review]
LGTM

**Comment by johngrantuk** [comment]
> Prettier applied a bunch of changes and made it a bit hard to follow actual changes I made 😕 Here's a list to make the review easier:
> 
>     * add buildCallWithPermit2 to addLiquidity section
> 
>     * add buildCallWithPermit to removeLiquidity section
> 
>     * add buildCallWithPermit2 to swap section
> 
>     * add Permit2 Helper section
> 
>     * add Permit Helper section
> 
>     * some other minor fixes
> 
> 
> Ps: I'm wondering why prettier auto-formatting was not applied before - how is this setup on your end?

Interesting, I didn't notice the prettier thing before. I didn't actually have prettier enabled at my end. Even when I do enable, it doesn't auto format on save. Does yours?

**Review by mkflow27 — APPROVED** [review]
Thanks Bruno.
